api/app/Http/Controllers/ChildEntityController.php

```diff
@@ -122,17 +122,17 @@ public function postOne($id, Request $request)
      * @param  Request $request
      * @return ApiResponse
      */
-    public function putOne($id, $childId, Request $request)
+    public function putOne(Request $request, $id, $childId)
     {
-        $model = $this->findParentEntity($id);
-        $childModel = $this->findOrNewChildEntity($childId, $model);
-        $validationRules = $this->getValidationRules();
-        if ($childModel->exists) {
-            $validationRules = $this->addIdOverrideValidationRule($validationRules, $childId);
-        }
-        $this->validateRequest($request->all(), $validationRules, $childModel->exists);
+        $parent = $this->findParentEntity($id);
+
+        $this->checkEntityIdMatchesRoute($request, $childId, $this->getChildModel());
+        $childModel = $this->findOrNewChildEntity($childId, $parent);
+
+        $this->validateRequest($request->all(), $this->addIdOverrideValidationRule($this->getValidationRules(), $childId));
```

Comment:
the same for other 3 places, to sum up it should be done for
`EntityController` and `ChildEntityController`
`putOne`, `patchOne`